### PR TITLE
fix(wallet): Remove Optimism from Ethereum Submenu

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Removes `Optimism` from the `Ethereum` network filters submenu.

Resolves <https://github.com/brave/brave-browser/issues/23181>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Make sure that `Test Networks` are enabled in brave://settings/wallet/networks
2. On the `Portfolio` page, click on the network filter and hover over `Ethereum`
3. `Optimism` should not be in the submenu.

Before:

https://user-images.githubusercontent.com/40611140/172688119-1ffd2eea-7ba0-4ee9-89e2-3ce8257eb0da.mov

After:

https://user-images.githubusercontent.com/40611140/172688147-9fb5f3bb-73d4-43f8-b5c3-5a794dbceef6.mov

### CODE REVIEW

**Review by josheleonard — APPROVED** [review]
LGTM

File: components/brave_wallet_ui/components/desktop/network-filter-selector/index.tsx [hunk 1]
```diff
@@ -95,7 +95,8 @@ function NetworkFilterSelector () {
                 <SubDropDown>
                   {sortedNetworks.filter((n) =>
                     n.coin === network.coin &&
-                    n.symbol.toLowerCase() === network.symbol.toLowerCase())
+                    n.symbol.toLowerCase() === network.symbol.toLowerCase() &&
+                    n.chainId !== BraveWallet.OPTIMISM_MAINNET_CHAIN_ID)
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/network-filter-selector/index.tsx:99]: maybe add a comment explaining why we hid this here?

Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/network-filter-selector/index.tsx:101]: Added a comment!

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.41.50 Chromium: 102.0.5005.99 (Official Build) nightly (64-bit) 
-- | --
Revision | `870f7bcc58dfa811cc68c2186439721385e086d0-refs/branch-heads/5005@{#1125}`
OS | Windows 11 Version 21H2 (Build 22000.708)

- Verified steps from PR
- Verified `Optimism` is not listed under Ethereum submenu anymore

https://user-images.githubusercontent.com/17010094/172905032-2b8db796-ad20-4f1c-8fbe-46d7af036602.mp4

**Pull request merged**